llvm/include/llvm/CodeGen/ResourceScoreboard.h

```diff
@@ -63,17 +63,21 @@ template <typename RC> class ResourceScoreboard {
   }
   RC &operator[](int Cycle) { return Cycles[(Head + Cycle) & (Size - 1)]; }
 
-  void reset(int D = 1) {
-    // Implementation relies on masking to wrap-around, so round up
-    // to a power of two.
-    int Pow2 = 1;
-    while (Pow2 < D) {
-      Pow2 += Pow2;
-    }
-    if (Cycles.empty()) {
+  void reset(int D = 0) {
```

Comment:
Nit: How about just introducing a `clear()` method?

Reply:
Yes, sure. Probably an unsupported hunch of many call sites to fix.